sdks/java/core/src/main/java/org/apache/beam/sdk/util/SerializableUtils.java

```diff
@@ -79,16 +83,31 @@ public static Object deserializeFromByteArray(byte[] encodedValue,
   }
 
   public static <T extends Serializable> T ensureSerializable(T value) {
-    @SuppressWarnings("unchecked")
-    T copy = (T) deserializeFromByteArray(serializeToByteArray(value),
-        value.toString());
-    return copy;
+    return clone(value);
   }
 
   public static <T extends Serializable> T clone(T value) {
+    final Thread thread = Thread.currentThread();
+    final ClassLoader tccl = ReflectHelpers.findClassLoader();
+    ClassLoader loader = tccl;
+    try {
+      if (tccl.loadClass(value.getClass().getName()) != value.getClass()) {
+        loader = value.getClass().getClassLoader();
+      }
+    } catch (final NoClassDefFoundError | ClassNotFoundException e) {
+      loader = value.getClass().getClassLoader();
+    }
+    if (loader == null) {
```

Comment:
You already set loader to tccl and all the other assignments never make this null again.

Reply:
tccl can be null, value.getClass().getClassLoader() can be null so 2 reasons for this null check ;)

Reply:
Your right.